modules/table/src/main/java/org/apache/ignite/internal/table/distributed/raft/PartitionCommandListener.java

```diff
@@ -57,10 +89,11 @@
             CommandClosure<WriteCommand> clo = iterator.next();
 
             if (clo.command() instanceof InsertCommand) {
-                BinaryRow previous = storage.putIfAbsent(
-                    extractAndWrapKey(((InsertCommand)clo.command()).getRow()),
-                    ((InsertCommand)clo.command()).getRow()
-                );
+                BinaryRow row = ((InsertCommand)clo.command()).getRow();
+
+                assert row.hasValue() : "Insert command should have a value.";
+
+                BinaryRow previous = storage.putIfAbsent(extractAndWrapKey(row), row);
```

Comment:
I didn't get the idea of `storage` in `PartitionCommandListener`. When this storage will be purged? Do we store all partitions data in this storage? Seems that this approach may lead to OOM. If this solution is temporary, where is todo with future changes?

Reply:
Added TODO and link to the issue.